File: smoke_tests/typescript/command_injection.ts
// Command Injection vulnerabilities in TypeScript
import { exec, spawn, execSync } from 'child_process';

class CommandInjectionVulnerabilities {
    executeCommand(userInput: string): void {
        // VULNERABLE: Command injection via exec
        exec(`ls ${userInput}`, (error, stdout, stderr) => {
            console.log(stdout);
        });
    }

    execSyncInjection(filename: string): string {
        // VULNERABLE: Command injection via execSync
        return execSync(`cat ${filename}`).toString();
    }

    spawnInjection(command: string): void {
        // VULNERABLE: Shell spawn with user input
        spawn('sh', ['-c', command]);
    }

    pingHost(host: string): void {
        // VULNERABLE: Command injection in ping
        exec(`ping -c 1 ${host}`);
    }

    executeScript(scriptPath: string): void {
        // VULNERABLE: Arbitrary script execution
        exec(`bash ${scriptPath}`);
    }

    grepLogs(pattern: string): void {
        // VULNERABLE: Command injection in grep
        exec(`grep "${pattern}" /var/log/app.log`);
    }

    evalInjection(code: string): any {
        // VULNERABLE: eval with user input
        return eval(code);
    }
}

export { CommandInjectionVulnerabilities };

File: smoke_tests/scala/command_injection.scala
// Command Injection vulnerabilities in Scala
import scala.sys.process._

class CommandInjectionVulnerabilities {
  def executeCommand(userInput: String): String = {
    // VULNERABLE: Command injection via Process
    s"ls $userInput".!!
  }

  def runShellCommand(command: String): String = {
    // VULNERABLE: Shell command execution
    Process(Seq("sh", "-c", command)).!!
  }

  def pingHost(host: String): String = {
    // VULNERABLE: Command injection in ping
    s"ping -c 1 $host".!!
  }

  def catFile(filename: String): String = {
    // VULNERABLE: Command injection via cat
    s"cat $filename".!!
  }

  def executeScript(scriptPath: String): Int = {
    // VULNERABLE: Arbitrary script execution
    s"bash $scriptPath".!
  }

  def grepLogs(pattern: String): String = {
    // VULNERABLE: Command injection in grep
    s"grep $pattern /var/log/app.log".!!
  }

  def processBuilderInjection(cmd: String): String = {
    // VULNERABLE: ProcessBuilder with user input
    Process(cmd).!!
  }
}
